File: frontend/tradion-web/src/app/core/status/job-status.ts
export const JobStatus = {
  draft: 'draft',
  open: 'open',
  inProgress: 'in progress',
  inProgressCompact: 'inprogress',
  completed: 'completed',
  done: 'done',
  closed: 'closed'
} as const;

export function normalizeJobStatus(status?: string | null): string {
  return (status || '').trim().toLowerCase();
}

export function isJobDraftLike(status?: string | null): boolean {
  return normalizeJobStatus(status) === JobStatus.draft;
}

export function isJobOpenLike(status?: string | null): boolean {
  return normalizeJobStatus(status) === JobStatus.open;
}

export function isJobInProgressLike(status?: string | null): boolean {
  const s = normalizeJobStatus(status).replace(/\s+/g, '');
  return s === JobStatus.inProgressCompact;
}

export function isJobCompletedLike(status?: string | null): boolean {
  const s = normalizeJobStatus(status);
  return s === JobStatus.completed || s === JobStatus.done || s === JobStatus.closed;
}

File: mobile/technician_app/lib/models/job_status.dart
class JobStatusValue {
  static const String draft = 'draft';
  static const String open = 'open';
  static const String inProgressSpaced = 'in progress';
  static const String inProgressCompact = 'inprogress';
  static const String completed = 'completed';
  static const String done = 'done';
  static const String closed = 'closed';

  static String norm(String? status) => (status ?? '').trim().toLowerCase();

  static bool isInProgressLike(String? status) {
    final s = norm(status).replaceAll(' ', '');
    return s == inProgressCompact;
  }

  static bool isCompletedLike(String? status) {
    final s = norm(status);
    return s == completed || s == done || s == closed;
  }
}
